assets/js/components/ChargingSessionModal.vue

```diff
@@ -140,42 +151,69 @@
 </template>
 
 <script>
-import Modal from "bootstrap/js/dist/modal";
 import "@h2d2/shopicons/es/regular/checkmark";
+import { distanceUnit, distanceValue } from "../units";
 import formatter from "../mixins/formatter";
+import Modal from "bootstrap/js/dist/modal";
 import api from "../api";
-import { distanceUnit, distanceValue } from "../units";
+
+import VehicleOptions from "./VehicleOptions.vue";
 
 export default {
 	name: "ChargingSessionModal",
+	components: { VehicleOptions },
 	mixins: [formatter],
 	props: {
 		session: Object,
+		vehicles: [Object],
 	},
-	emits: ["session-deleted"],
+	emits: ["session-changed"],
 	methods: {
 		openSessionDetailsModal() {
 			const modal = Modal.getOrCreateInstance(document.getElementById("sessionDetailsModal"));
 			modal.show();
 		},
-		confirmRemoving() {
+		openRemoveConfirmationModal() {
 			const modal = Modal.getOrCreateInstance(
 				document.getElementById("deleteSessionConfirmationModal")
 			);
 			modal.show();
 		},
-		async removeSession() {
+		formatKm: function (value) {
+			return `${distanceValue(value)} ${distanceUnit()}`;
+		},
+		async changeVehicle(index) {
+			await this.updateSession({
+				vehicle: this.vehicles[index - 1].title,
+			});
+		},
+		async removeVehicle() {
+			await this.updateSession({
+				vehicle: this.$t("main.vehicle.unknown"),
```

Comment:
Das führt zu komischen Dingen. Es gibt ja technisch kein Fahrzeug was "Gastfahrzeug" heißt in der Datenbank. Hiermit schreiben wir den Übersetzungstext der aktuell in der UI ausgewählten Sprache in die Datenbank. Nach einem Sprachwechsel (oder Änderung des Texts durch ein evcc Update) passt das dann nicht mehr zueinander. Das Ändern auf "Gastfahrzeug" muss dazu führen, dass in der DB das `vehicle` Feld wieder `nil` ist.

Das beißt sich etwas mit deinem Kommentar @andig, dass wir ein generisches PUT haben wollen was nur alle `nil` Werte ändert. Ich wäre weiterhin dafür Fahrzeugänderung als expliziten Case zu modellieren:

``` 
PUT /sessions/<id>/vehicle
DELETE /sessions/<id>/vehicle
```

Zudem macht dann auch gleich die API Dokumentation klar was änderbare Werte sind und was nicht. Gerade in der Session sehe ich auch nicht, dass wir die anderen Werte überschreibbar machen wollen. Später evtl. zusätzliches Labeling oder so. Aber das wäre für mich ein anderes Thema.

Reply:
Das Problem hat mich bei der Implementierung auch schon beschäftigt. Wechselt man eine Sprache, dann würde ja das gewählte Auto auch ein anderes werden.
Ich bin gespannt auf @andig 's Meinung zu deinem Lösungsvorschlag.

Reply:
Ich habe eine Lösung gefunden, bei der wir es generisch halten können und dennoch selektiert Daten ändern können. Auch zu `nil`

Reply:
Schaut euch das gerne mal an, wie es jetzt implementiert ist.